### PR TITLE
Pods 7189

### PR DESCRIPTION
**Pull request by shehzad-ismail8**

### CODE REVIEW

**Review by ngmjohnstone — APPROVED** [review]
Looks good to me

File: app/services/EventReportService.scala [hunk 1]
```diff
@@ -102,42 +108,41 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
...
-      case Right(true) =>
-        eventReportConnector.compileEventReportSummary(pstr, data).map { response =>
-          Ok(response.body)
-        }
-      case Left(errors) =>
-        val allErrorsAsString = "Schema validation errors:-\n" + errors.mkString(",\n")
-        throw EventReportValidationFailureException(allErrorsAsString)
-      case _ => throw EventReportValidationFailureException("compileEventReportSummary schema validation failed (returned false)")
-    }
+    onValidJason(pstr, data, createCompiledEventSummaryReportSchemaPath, "compileEventReportSummary")(
```
Comment (cyeniceri) [inline app/services/EventReportService.scala:111]: Can you refactor method name please ?

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: app/services/EventReportService.scala (hunk 1 above)
Comment (shehzad-ismail8) [reply app/services/EventReportService.scala]: Changed name to onValidJson

File: scalastyle-config.xml [hunk 2]
```diff
@@ -103,7 +103,7 @@
  </check>
  <check level="warning" class="org.scalastyle.scalariform.MethodNamesChecker" enabled="true">
   <parameters>
-   <parameter name="regex"><![CDATA[^[a-z][A-Za-z]*$]]></parameter>
```
[the remark this answers is not in the document]
Comment (shehzad-ismail8) [reply scalastyle-config.xml:106]: This was to include numbers in the method name for the new event type

**Comment by cyeniceri** [comment]
We're adding a new endpoint here, so let's add this new endpoint into postman-collection as a new request.
https://github.com/hmrc/pension-scheme-event-reporting-postman

File: app/services/EventReportService.scala (hunk 1 above)
Comment (shehzad-ismail8) [reply app/services/EventReportService.scala]: Changed name to validatePayload

**Pull request merged**